cmd/snap-confine/udev-support.c

```diff
@@ -304,4 +305,5 @@ void setup_devices_cgroup(const char *security_tag, struct snappy_udev *udev_s)
 		run_snappy_app_dev_add(udev_s, path);
 		udev_s->assigned = udev_list_entry_get_next(udev_s->assigned);
 	}
+	(void)sc_set_effective_identity(old);
```

Comment:
I think this can be moved all the way up to after the mkdir(), no? If not, please add a supporting comment.

Reply:
I think it cannot because `run_snappy_app_dev_add` was again making stuff as user group. I'll double check and add all the comments.

Reply:
Aha, I think you were right and we don't create anything new per se (we do open existing files inside the cgroup code but those are never made by us). I've changed this locally and will commit and push after it passes the spread test added in this branch.

Reply:
Confined and pushed.